components/clp-package-utils/clp_package_utils/scripts/start_clp.py

```diff
@@ -730,6 +731,83 @@ def start_webui(instance_id: str, clp_config: CLPConfig, mounts: CLPDockerMounts
     logger.info(f"Started {component_name}.")
 
 
+def get_host_ip():
+    s = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+    s.settimeout(0)
+    try:
+        # doesn't even have to be reachable
+        s.connect(("10.254.254.254", 1))
+        ip = s.getsockname()[0]
+    except Exception:
+        ip = "127.0.0.1"
+    finally:
+        s.close()
+    return ip
+
+
+def start_reducer(
+    instance_id: str,
+    clp_config: CLPConfig,
+    container_clp_config: CLPConfig,
+    num_cpus: int,
+    mounts: CLPDockerMounts,
+):
+    component_name = REDUCER_COMPONENT_NAME
+    logger.info(f"Starting {component_name}...")
+
+    container_name = f"clp-{component_name}-{instance_id}"
+    if container_exists(container_name):
+        logger.info(f"{component_name} already running.")
+        return
+
+    container_config_filename = f"{container_name}.yml"
+    container_config_file_path = clp_config.logs_directory / container_config_filename
+    with open(container_config_file_path, "w") as f:
+        yaml.safe_dump(container_clp_config.dump_to_primitive_dict(), f)
+
+    logs_dir = clp_config.logs_directory / component_name
+    logs_dir.mkdir(parents=True, exist_ok=True)
+    container_logs_dir = container_clp_config.logs_directory / component_name
+
+    clp_site_packages_dir = CONTAINER_CLP_HOME / "lib" / "python3" / "site-packages"
+    # fmt: off
+    container_start_cmd = [
+        'docker', 'run',
+        '-di',
+        '--network', 'host',
+        '-w', str(CONTAINER_CLP_HOME),
+        '--name', container_name,
+        '-e', f'PYTHONPATH={clp_site_packages_dir}',
+        '-e', f'CLP_LOGS_DIR={container_logs_dir}',
+        '-e', f'CLP_LOGGING_LEVEL={clp_config.reducer.logging_level}',
+        '-e', f'CLP_HOME={CONTAINER_CLP_HOME}',
+        '--mount', str(mounts.clp_home),
+    ]
+    #fmt on
+    necessary_mounts = [
+        mounts.logs_dir,
+    ]
+    for mount in necessary_mounts:
+        if mount:
+            container_start_cmd.append('--mount')
+            container_start_cmd.append(str(mount))
+    container_start_cmd.append(clp_config.execution_container)
+
+    reducer_cmd = [
+        'python3', '-u', '-m',
+        'job_orchestration.reducer.reducer',
+        '--config', str(container_clp_config.logs_directory / container_config_filename),
+        '--host', get_host_ip(),
```

Comment:
Ideally, we'd allow the user to choose which interface they want to bind to in the config file (like we do for other components). I know that might be less convenient than just copying the package to whatever machine you want and running it, but if anyone is not running on localhost, then they likely should be spending some time to configure things correctly anyway.

Reply:
Yeah you're right, I'll change it to rely on config